include/GoogleSync/GoogleSyncBase.php

```diff
@@ -0,0 +1,1049 @@
+        $event_remote_empty = new Google_Service_Calendar_Event;
+
+        $extendedProperties = new Google_Service_Calendar_EventExtendedProperties;
+        $extendedProperties->setPrivate(array());
+
+        $event_remote_empty->setExtendedProperties($extendedProperties);
+
+        //Set the Google Event up to match the SuiteCRM one
+        $event_remote = $this->updateGoogleCalendarEvent($event_local, $event_remote_empty);
+
+        return $event_remote;
+    }
+
+    /**
+     * Set the timezone
+     *
+     * @param string $timezone : timezone_identifier (ie. America/New_York)
+     *
+     * @return bool Success/Failure
+     */
+    protected function setTimezone($timezone)
+    {
+        if (date_default_timezone_set($timezone)) {
+            $this->timezone = date_default_timezone_get();
+            $this->logger->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Timezone set to \'' . $this->timezone . '\'');
+            return true;
+        }
+        // TODO: I think a warning enough into log instead fatal (but it should be handled by caller function)
+        $this->logger->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Failed to set timezone to \'' . $timezone . '\'');
+        return false;
+    }
+
+    /**
+     * Set the last sync time for the record
+     *
+     * This *must* be called *after* the sync is done
+     * This also saves the event, so you don't need to do it twice. Just call this.
+     * Also adds the id of the event so it doesn't get synced again this session.
+     *
+     * @param Meeting $event_local SuiteCRM Meeting bean
+     * @param string $gEventId (optional) The ID that Google has for the event.
+     *
+     * @return string|bool Meeting Id on success, false on failure
+     */
+    protected function setLastSync(Meeting $event_local, $gEventId = null)
+    {
+        if (isset($gEventId)) {
+            $event_local->gsync_id = $gEventId;
+        }
+
+        $event_local->gsync_lastsync = time() + 3; // we add three seconds to this, so that the modified time is always older than the gsync_lastsync time
+        $return = $event_local->save(false);
+
+        $isValidator = new SuiteValidator();
+        if ($isValidator->isValidId($return)) {
+            if (!$this->workingUser instanceof User) {
+                throw new GoogleSyncException('GoogleSyncBase is trying to setLastSync but workingUser type is incorrect, ' . gettype($this->workingUser) . ' given.', GoogleSyncException::INCORRECT_WORKING_USER_TYPE);
+            }
+            $event_local->set_accept_status($this->workingUser, 'accept');  // Set the meeting as accepted by the user, otherwise it doesn't show up on the calendar. We do it here because it must be saved first.
+            //$this->syncedList[] = $event_local->id;
+            return $event_local->id;
+        }
+        // TODO: I think a warning enough into log instead fatal (but it should be handled by caller function)
```

Comment:
@gymad 
If it gets here, something bad happened. I'll find an appropriate way to handle it.

Reply:
Fixed